### ISSUE TITLE
the loss always NAN

### ISSUE THREAD

**Issue by foolishcqx** [WindFromWestAndHairToEast/WavThruVec_pytorch#1]
![Uploading 微信图片_20240421144204.png…]()
Epoch[192/2001],step [202000/211200]:W2V Feat Loss: nan, W2V Feat PostNet Loss: nan,attn binarization loss:nan,Current Learning Rate is 0.100000.Time Used: 142124.986s,Estimated Time Remaining: 6594.002s.

**Comment by p1an-lin-jung** [comment]
Yes, I also encountered this problem last year, but in the past few months, I have been busy with other projects. 
I made some modifications at that time, but they were not updated on GitHub.  If I have free time recently, I will handle this issue.

At the same time, you should also check the MFA version and MFA's output. Because I found that higher versions of MFA would treat silent fragments as "", while in lower versions, they are "sp", "sli", and "spn". This can cause some bugs when you use `tgt. io. readutextgrid` to read TextGrid files.

In addition, if the duration of a phoneme is too short (e.g. 0.01 seconds), it will be calculated as 0 frames. This case may also trigger NAN loss.  You can regenerate the alignment file (because the MFA‘s output may differ slightly each time), or remove the problematic files in this section

**Comment by p1an-lin-jung** [comment]
It is worth mentioning that I remember Text2vec was able to work properly, while the vocoder Vec2wav was not.

Here are my guesses on possible reasons: due to insufficient CUDA memory, I set the batch size very small when training vec2wav. However, vec2wav uses the `ConditionalBatchNorm1d` module, so a small batch size can have a negative impact on the model's performance. 

If I have time recently, I will train again on a machine with more CUDA memory.

**Comment by foolishcqx** [comment]
Thanks for the answer, I will try it based on your suggestion
